draft-ietf-jose-hpke-encrypt.md

```diff
@@ -347,11 +353,17 @@ HPKE also offers modes that offer authentication.
 HPKE relies on a source of randomness to be available on the device.
 In Key Agreement with Key Wrapping mode, CEK has to be randomly generated and it MUST be ensured that the guidelines in {{RFC8937}} for random number generations are followed.
 
+## HPKE authentication
+
+Authenticated HPKE modes MUST NOT be used for Key Encryption, as they do not authenticate the message. These modes validate the sender's identity by ensuring that the shared secret originates from the sender's static public key, a pre-shared key, or a combination of both. However, this authentication is limited to the key exchange process and does not cover the message encrypted with the CEK. For example, a malicious recipient could decrypt the CEK, modify the decrypted message, and re-encrypt it using the same CEK. Other recipients, unaware of the tampering, might mistakenly trust the altered message as if it came from the authenticated sender.
+
 ## Key Management
 
-Reusing a single KEM key across multiple algorithm combinations MUST be avoided to maintain cryptographic security. Each key and its associated algorithm suite, comprising the KEM, KDF, and AEAD, should be managed independently. This separation prevents unintended interactions or vulnerabilities between suites, ensuring the integrity and security guarantees of each algorithm suite are preserved.
-Additionally, the same key MUST NOT be used for both key wrapping and content encryption, as it may introduce security risks.  It
-creates algorithm confusion, increases the potential for key leakage, cross-suite attacks, and improper handling of the key.
+A single key MUST NOT be used in both sender and recipient roles. Avoiding the use of the same key for both sender and recipient roles ensures clear cryptographic boundaries and minimizes unintended interactions.
```

Comment:
This prevents encrypting messages to yourself, which part of the HPKE specification makes this requirement?

Reply:
Encrypting messages to yourself is typically done using symmetric algorithms, For instance, in case of PQC KEM, it looks complex to encrypt the symmetric key to yourself using encap and then later using decap to get the symmetric key. I don't see any benefit for the additional compute.

Reply:
I'm not sure why the good guidance above is being deleted.

Reply:
As per our discussion in the call, reverted the text, it will take up in a separate PR.